### DISCUSSION TITLE
Yes, we're still alive.

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2131] [General]
We appreciate all of the questions about our well being. We're all definitely still alive. Believe me.

An update is definitely coming.

**Comment by huntingm27** [comment]
Just a honest question do you know a rough estimate of when tshock 1.4.1 will come out

**Comment by ghost** [comment]
i have quesiton, i saw on Tshock website that running tshock server on my main pc or system idk is dangerous and i need o use chroot jail linux. but i dont have linux what can i do

**Reply by rautamiekka** [reply]
Paravirtualizer like VMware Workstation Player or Oracle VirtualBox. You'll need a 64-bit CPU with Intel VT or AMD-V hardware virtualization support, and the CPU power and the RAM cuz a paravirtualizer is virtualizing another OS, which ain't a cheap process. But you need to modify the firewall and maybe forward ports unless you use ZeroTier One.

Or pay for a cheap VPS that has the specs to run the Server.

**Reply by ghost** [reply]
> Paravirtualizer like VMware Workstation Player or Oracle VirtualBox. You'll need a 64-bit CPU with Intel VT or AMD-V hardware virtualization support, and the CPU power and the RAM cuz a paravirtualizer is virtualizing another OS, which ain't a cheap process. But you need to modify the firewall and maybe forward ports unless you use ZeroTier One.
> 
> Or pay for a cheap VPS that has the specs to run the Server.

so it.s unsafe to run tshock on main machine

**Reply by rautamiekka** [reply]
^ I've no idea about whether it's unsafe like some ppl say TeamSpeak 3 Server is. TTTT I don't even have any ideas why TShock would be unsafe (while TS3 is (or was, I haven't looked into it for months) using Qt version which is proven to be seriously broken security-wise).

I simply gave the best options while assuming you don't have a machine you can dedicate for hosting TShock on Linux.

**Reply by ghost** [reply]
> ^ I've no idea about whether it's unsafe like some ppl say TeamSpeak 3 Server is. TTTT I don't even have any ideas why TShock w

would it work with vm like if i just drop the terraria world somewhere? or need i download steam

**Reply by rautamiekka** [reply]
^ If you already know how TShock works it's in no way diff from running inside a VM, that's the whole point.

The world files are completely self-contained/portable. TShock doesn't need 'Terraria' nor Steam, other than the world file.

**Reply by hakusaro** [reply]
1. Whoever said TShock is dangerous to run on your main machine is literally completely uninformed.
2. TShock doesn't need Steam.
3. Please create new discussions, don't just reply to existing ones.

**Reply by hakusaro** [reply]
Note: You still should run as a separate user, and remember that we're mostly running the Terraria binary. So, therefore, if you run Terraria on your main machine, it's mostly the same code in TShock.

**Comment by Syshein** [comment]
Hey guys,

first of all: You are doing a great job :) 

I know it's a taboo to ask for your activities on this repo but can you tell me something about the current state of development? A few friends of mine and I would like to play the new content of Terraria 1.4(.1) and I need to host a server for them. But i can't find any info wether all new items, bosses and stuff are supported by the current tshock version. It would be really frustrating for all of us if I set up a server just to find that the new stuff is still unsupported. 

I really don't want to rush anyone so no matter if the new things are working already or not. thanks for all the great work you are doing.
